### PR TITLE
Some more profile page work

### PR DESCRIPTION
**Pull request by rob-3**
- Make name properly optional
- Add profilePage to navigation
- Add initial ProfilePage skeleton
- Add spacing around views and saved
- Make some improvements to searchbar cross-platform
- Get "grid" layout roughly working
- Add ability to switch between created and collections views
- Show username
- Get image and collection counts working

### CODE REVIEW

File: mobile/src/pages/ProfilePage.tsx [hunk 1]
```diff
@@ -1,37 +1,157 @@
...
+  const {
+    data: user,
+    isLoading: isProfileLoading,
+    refetch: refetchProfile,
+  } = useQuery({
+    // FIXME figure out how to incorporate userid here
+    queryKey: ["user_profile"],
+    queryFn: async () => {
+      const userId = (await getAuthSession()).userId;
+      return getUser({ userId });
```
Comment (kitorer) [inline mobile/src/pages/ProfilePage.tsx:41]: ohh btw, you dont actually need to get userId there. the getUser() should be getting it for you as long as you're logged in
L39 https://github.com/poop-team/daydream/blob/main/mobile/src/requests/fetch.ts
```
export async function getUser({ userId = "", userName = "" }) {
  const params = new URLSearchParams({
    userId: (await getAuthSession()).userId,
    searchUserId: userId || (await getAuthSession()).userId,
    userName: userName,
  });```

File: mobile/src/pages/ProfilePage.tsx [hunk 2]
```diff
@@ -1,37 +1,157 @@
...
+import { useEffect, useState } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { getAuthSession } from "../utils/storage";
+import { getUser } from "../requests/fetch";
+import { ErrorResponse } from "../types/error.type";
+import useInfiniteQueryPosts from "../hooks/useInfiniteQueryPosts";
+import { Post } from "../types/post.type";
 
 export default ({ navigation }) => {
   const [search, setSearch] = useState("");
```
Comment (kitorer) [inline mobile/src/pages/ProfilePage.tsx:14]: its useDebounce() time

**Review by kitorer — CHANGES_REQUESTED** [review]

File: mobile/src/pages/ProfilePage.tsx (hunk 1 above)
Comment (rob-3) [reply mobile/src/pages/ProfilePage.tsx:41]: Good to know, I will remove this

File: mobile/src/pages/ProfilePage.tsx (hunk 2 above)
Comment (rob-3) [reply mobile/src/pages/ProfilePage.tsx:14]: Yep

**Review by kitorer — APPROVED** [review]

**Pull request merged**